gprofiler/profilers/node.py

```diff
@@ -94,80 +106,77 @@ def _get_debugger_url() -> str:
     return cast(str, response_json[0]["webSocketDebuggerUrl"])
 
 
-def _send_socket_request(sock: WebSocket, cdp_request: Dict) -> None:
-    sock.send(json.dumps(cdp_request))
-    message = sock.recv()
-    try:
-        message = json.loads(message)
-    except json.JSONDecodeError:
-        raise NodeDebuggerUnexpectedResponse(message)
-
-    if (
-        "result" not in message.keys()
-        or "result" not in message["result"].keys()
-        or "type" not in message["result"]["result"].keys()
-        or message["result"]["result"]["type"] != "boolean"
-    ):
-        raise NodeDebuggerUnexpectedResponse(message)
-
-
-def _execute_js_command(sock: WebSocket, command: str) -> Any:
+@retry(NodeDebuggerProcessUndefined, 5, 0.5)
+def _evaluate_js_command(sock: WebSocket, command: str, expected_result: str) -> Any:
+    # Check if process or process.mainModule in command, and if it is, check if it is defined in js context
+    if "process.mainModule" in command:
+        command = f'typeof(process.mainModule) === "undefined" ? "process undefined" : {command}'
+    if "process" in command:
+        command = f'typeof(process) === "undefined" ? "process undefined" : {command}'
     cdp_request = {
         "id": 1,
         "method": "Runtime.evaluate",
         "params": {
             "expression": command,
-            "replMode": True,
         },
     }
     sock.send(json.dumps(cdp_request))
     message = sock.recv()
     try:
         message = json.loads(message)
     except json.JSONDecodeError:
-        raise NodeDebuggerUnexpectedResponse(message) from None
+        if expected_result == ResultType.NONE:
```

Comment:
Do you expect a specific value if `ResultType.NONE`?

Such code is not very clear - IDK why `ResultType.NONE` would yield `JSONDecodeError`, I think it's nicer if you check for the value you expct for `None` and handle it. Otherwise, there could be calls with ResultType.NONE`, which actually result in unexpected, non-JSON output, and you'll ignore it here

Reply:
When evaluating _debugEnd() JS debugger returns empty message == '' thats why it yields `JSONDecodeError`. I can add check `if len(message) == 0 and expected_result == ResultType.NONE`

Reply:
Yes, that's better